packages/config/scripts/omnichain/coingecko/index.ts

```diff
@@ -0,0 +1,202 @@
+  balance: bigint
+  price: number
+  decimals: bigint
+  bridgedValueUsd: bigint
+}
+
+async function compareWithLive(
+  bridgedValues: BridgedValue[],
+  totalBridgedValueUsd: bigint,
+) {
+  // TODO: agEUR is not included in the total value as the token that is bridged is a wrapped version of it
+  // so coingecko doesn't have the price for it. This
+  const THRESHOLD_USD = 10_000_000n
+
+  const response = await fetch(
+    'https://api.l2beat.com/api/tvl/aggregate?projectSlugs=omnichain',
+  )
+  const apiResponse = (await response.json()) as {
+    hourly: { data: [number, number][] }
+  }
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const liveValueUsd = BigInt(Math.floor(apiResponse.hourly.data.at(-1)![1]))
+  assert(
+    totalBridgedValueUsd > liveValueUsd,
+    'Total value is less than live value. Something is wrong.',
+  )
+
+  const diff = totalBridgedValueUsd - liveValueUsd
+
+  if (diff > THRESHOLD_USD) {
+    console.log(
+      `Total value is ${diff} greater than live value. Probably a new token appeared. Check the new tokens that appeared in tokens.json`,
+    )
+
+    const dataToSave = {
+      totalBridgedValueUsd: totalBridgedValueUsd,
+      tokens: bridgedValues.sort((a, b) => a.address.localeCompare(b.address)),
+    }
+
+    writeFileSync(
+      './scripts/omnichain-value/tokens.json',
+      JSON.stringify(
+        dataToSave,
+        (_, value) =>
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+          typeof value === 'bigint' ? value.toString() : value,
+        2,
+      ),
+    )
+  } else {
+    console.log(
+      `Total value is ${diff} greater than live value. This is within the ${THRESHOLD_USD} threshold. No action required.`,
+    )
+  }
+}
+
+async function calculateBridgedValue(
+  token: {
+    address: string
+    name: string
+    symbol: string
+    coingeckoId: CoingeckoId
+  },
+  provider: providers.AlchemyProvider,
+  prices: Record<string, { usd: number }>,
+): Promise<BridgedValue | undefined> {
+  const contract = new Contract(token.address, ERC20_ABI, provider)
+
+  const [balance, decimals] = await Promise.all([
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+    ((await contract.balanceOf(token.address)) as BigNumber).toBigInt(),
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+    BigInt((await contract.decimals()) as number),
+  ])
+
+  if (balance === 0n) {
+    return
+  }
+
+  const price = prices[token.coingeckoId.toString()].usd
+  const priceBn = BigInt(Math.floor(price * 10 ** 18))
+  return {
+    ...token,
+    balance,
+    price,
+    decimals,
+    bridgedValueUsd: (balance * priceBn) / (BigInt(10 ** 18) * 10n ** decimals),
+  }
+}
+
+function filterTokenAddresses(
+  coinList: {
+    symbol: string
+    name: string
+    platforms: Record<string, string | null>
+    id?: CoingeckoId
+  }[],
+  addresses: string[],
+) {
+  const tokenAddresses = coinList
```

Comment:
This variable name is deceiving, it's more than just addresses. This naming goes back to `index.ts` as well